### PR TITLE
docs: generate a website with the specs

### LINKED ISSUE #260: Automatically generate docs for the gherkin scenarios
Docs: http://docs.picklesdoc.com/en/latest/StartSimple/
Docker image: https://github.com/Xeli/picklesdoc-docker/blob/master/Dockerfile (we could push it to our registry)
Sample run: 
```shell
docker run --rm -it -v $(pwd):/project docker.elastic.co/observavbility-ci/picklesdoc:latest \
           -f /project/features -o /project/docs --sn "E2E Fleet" --sv 1.0.0
```

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
This PR generates a website with the HTML representation of the gherkin feature files, so that it's easier to consume them.

The generation is done using picklesdoc (http://docs.picklesdoc.com), a Windows application that parses Gherkin files and generates an HTML output. We have wrapped the Windows binary into a Docker image (using mono), and pushing this image to our registry:

- docker.elastic.co/observavbility-ci/picklesdoc:2.21.0 (Dockerfile here: https://github.com/Xeli/picklesdoc-docker/blob/master/Dockerfile)

Because this image uses a Windows binary, we wrapped the execution into a Make target, which massages the output to remove bad slashes in the generated paths (i.e. the root index.html file).

## Why is it important?
We want to provide an easy way to consume the specification, shareable and understandable by anybody in the team, and HTML is easy to read.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests for the CLI, and they are passing locally
- [ ] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Author's Checklist

- [x] Generated the docs locally

## How to test this PR locally

```shell
$ cd e2e
$ make build-docs
```

A website will be generate under the `docs` directory

## Related issues

- Closes #260 

## Screenshots

Sample website:

Screenshot 2020-09-03 at 17 38 49

## Logs

```
Generating documentation based on the following parameters 
---------------------------------------------------------- 
Feature Directory              : /suites 
Output Directory               : /suites/docs 
Project Name                   : E2E Testing 
Project Version                : 1.0.0 
Language                       : en 
Incorporate Test Results?      : No 
Include Experimental Features? : No 
Exclude Tag                    :  
Technical Tag                    :  
Writing HTML to /suites/docs 
Pickles completed successfully
```

## Follow-ups
The templates are embedded into the binary, need to research about customising the HTML templates.

We also need to put the Dockerfile for the picklesdoc image in a repo.

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -136,6 +136,28 @@ pipeline {
...
+          }
+          steps {
+            deleteDir()
+            unstash 'source'
+            dir("${BASE_DIR}/e2e") {
+              sh(label: 'Build docs', script: 'make build-docs')
+            }
+            post {
+              always {
+                archiveArtifacts allowEmptyArchive: true, artifacts: "${BASE_DIR}/e2e/docs/**"
```
Comment (v1v) [inline .ci/Jenkinsfile:156]: nit: cosmetic change, if it helps to access to the URL with a few less folders

```suggestion
                dir("${BASE_DIR}") {
                  archiveArtifacts allowEmptyArchive: true, artifacts: "e2e/docs/**"
                }
```

**Review by v1v — APPROVED** [review]

**Review by cachedout — APPROVED** [review]

**Comment by mdelapenya** [comment]
@EricDavisX you'll love this: https://beats-ci.elastic.co/job/e2e-tests/job/e2e-testing-mbp/job/PR-261/11/artifact/e2e/docs/_suites/ingest-manager/features/fleet_mode_agent.html

**Comment by mdelapenya** [comment]
The errors are not related, as the agent structure has changed

**Comment by EricDavisX** [comment]
excellent indeed!

**Pull request merged**